ntoskrnl/cc/view.c

```diff
@@ -75,13 +75,16 @@ ULONG CcRosVacbIncRefCount_(PROS_VACB vacb, PCSTR file, INT line)
 ULONG CcRosVacbDecRefCount_(PROS_VACB vacb, PCSTR file, INT line)
 {
     ULONG Refs;
+    BOOLEAN VacbDirty = vacb->Dirty;
+    BOOLEAN VacbTrace = vacb->SharedCacheMap->Trace;
+    BOOLEAN VacbPageOut = vacb->PageOut;
 
     Refs = InterlockedDecrement((PLONG)&vacb->ReferenceCount);
-    ASSERT(!(Refs == 0 && vacb->Dirty));
-    if (vacb->SharedCacheMap->Trace)
+    ASSERT(!(Refs == 0 && VacbDirty));
+    if (VacbTrace)
     {
         DbgPrint("(%s:%i) VACB %p --RefCount=%lu, Dirty %u, PageOut %lu\n",
```

Comment:
VacbDirty and VacbPageOut are BOOLEANs aka. one-byte long. Not sure using %u would correctly work there.

Reply:
"%i" and "%d" are the same thing, and "%u" and "%lu" are also the same thing on Windows. And it doesn't matter whether the parameter is BOOLEAN or char or USHORT, it will always be a 32 bit integer on the stack.

Reply:
For sure, but then what's going to be printed on screen? since only the low byte is being defined...

Reply:
Everything is default promoted to int, so FALSE is printed as 0, TRUE is printed a 1.